packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/analytics_client.dart

```diff
@@ -0,0 +1,143 @@
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/key_value_store.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/session_manager.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/stoppable_timer.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/sdk/pinpoint.dart';
+import 'package:amplify_core/amplify_core.dart';
+
+/// Top level client for executing all behaviors of the Analytics Plugin
+/// by managing behaviors and inter dependencies of sub clients and helper classes
+class AnalyticsClient {
+  AnalyticsClient._(
+    this._sessionManager,
+    this._eventClient,
+    this._eventCreator,
+    this._endpointClient,
+  );
+  final SessionManager _sessionManager;
+  final EventClient _eventClient;
+  final EventCreator _eventCreator;
+  final EndpointClient _endpointClient;
+
+  /// TODO - clarify - what is behavior on app refresh / hot reload?
+  late final StoppableTimer _autoEventSubmitter = StoppableTimer(
+    duration: const Duration(seconds: 10),
+    function: (Timer t) => flushEvents(),
+  );
+
+  /// Use parameters to initialize sub clients and helper classes
+  static Future<AnalyticsClient> getInstance(
+    String appId,
+    KeyValueStore keyValueStore,
+    PinpointClient pinpointClient,
+    CachedEventsPathProvider? pathProvider,
+    AppLifecycleProvider? appLifecycleProvider,
+    DeviceContextInfo? deviceContextInfo,
+  ) async {
+    final eventCreator =
+        await EventCreator.getInstance(keyValueStore, deviceContextInfo);
+
+    final endpointClient = await EndpointClient.getInstance(
+      appId,
+      keyValueStore,
+      pinpointClient,
+      deviceContextInfo,
+    );
+
+    final eventClient = EventClient(
+      appId,
+      keyValueStore,
+      endpointClient,
+      pinpointClient,
+      pathProvider,
+    );
+
+    // Initialize session manager
+    // And define auto session start/end Events to be sent on session start/end events
+    final sessionManager = SessionManager(
+      keyValueStore,
+      appLifecycleProvider,
+      onSessionEnd: (sb) {
+        eventClient.recordEvent(
+          eventCreator.createPinpointEvent(sessionStopEventType, sb),
+        );
+        eventClient.flushEvents();
+      },
+      onSessionStart: (sb) async {
+        await endpointClient.updateEndpoint();
+        eventClient.recordEvent(
+          eventCreator.createPinpointEvent(sessionStartEventType, sb),
+        );
+      },
+    );
+
+    return AnalyticsClient._(
+      sessionManager,
+      eventClient,
+      eventCreator,
+      endpointClient,
+    );
+  }
+
+  Future<void> flushEvents() async {
+    await _eventClient.flushEvents();
+  }
+
+  /// Create a [Event] object from [AnalyticsEvent] received from the public API
+  /// Received events are NEVER sent immediately but cached to be sent in a batch
```

Comment:
Thanks let's also keep this open for now in case other people encounter this confusion. 

Updated comments to :

  /// Create a Pinpoint [Event] object from [AnalyticsEvent] received from the public API.

Reply:
Yeah agreed, it's surprising how we don't have comments on that.